dvc/repo/list_outs.py

```diff
@@ -0,0 +1,63 @@
+import logging
+import os
+
+
+logger = logging.getLogger(__name__)
+
+
+@staticmethod
+def list_outs(url, targets=[], rev=None, recursive=None):
+    if os.path.exists(url):
+        path = os.path.abspath(url)
+        return _list_outs_local_repo(path, targets, recursive)
+    else:
+        return _list_outs_git_repo(url, targets, rev, recursive)
+
+
+def _list_outs_local_repo(path, targets=[], recursive=None):
+    from dvc.repo import Repo
+    from dvc.path_info import PathInfo
+
+    repo = Repo(path)
+
+    outs = set()
+    for stage in repo.stages:
+        outs = outs.union(stage.outs)
+
+    matched_outs = []
+    if len(targets):
+
+        def func(out, path_info, is_dir):
+            if out.scheme == "local" and out.path_info.__eq__(path_info):
+                return True
+
+            if is_dir:
+                if recursive and out.path_info.isin(path_info):
+                    return True
+                elif out.path_info.isin(path_info, 1):
+                    return True
+
+            return False
+
+        for path in targets:
+            abs_path = os.path.abspath(path)
+            path_info = PathInfo(abs_path)
+            is_dir = os.path.isdir(abs_path)
```

Comment:
You could `dvc add dir/`, so it might not exist here(because there was no `dvc checkout` in that cloned repo yet). We need to handle that case as well.

Reply:
Could you please provide me more info about the case, I cannot understand it correctly
What steps should I proceed for reproducing behavior
There are mine:
```
$ mkdir add_dir
$ dvc add add_dir/                                                             
WARNING: 'add_dir' is empty.                                                                                                                           
....                                                                                                                                                   
To track the changes with git, run:                                                                                                                    
                                                                                                                                                       
        git add add_dir.dvc .gitignore
$ python -m dvc list  .
auc.metric
data/data.xml
data/features
add_dir
model.pkl
data/prepared
```

Reply:
@JIoJIaJIu Your example is almost right, except that after `dvc add add_dir` you could delete `add_dir`(but not add_dir.dvc) and `dvc list .` should still work and show `add_dir`. For example say you have this git repo on github, and you do `dvc list https://github.com/me/myproj`, then `add_dir` won't exist in git, so your `os.path.isdir()` will fail, because we don't `dvc checkout` in the cloned repo(it should stay that way), so you can't rely on os.path when dealing with dvc-tracked dirs. Let me know if that makes sense.